drivers/libusb1.c

```diff
@@ -1090,7 +1090,7 @@ static int nut_libusb_get_interrupt(
 		) {
 			return -1;
 		}
-		ret = (usb_ctrl_charbufsize)bufsize;
+		ret = (usb_ctrl_charbufsize)tmpbufsize;
 	}
```

Comment:
For the question in PR ("can we trust libusb...") -- I think the if-clause just above range-checks the `tmpbufsize` to fit in `usb_ctrl_charbufsize` range, so any problems should be caught there.

Reply:
oh yes you're right, the || ensures that condition is also evaluated, I missed that.